Packs/DevSecOps/Integrations/GitLab/GitLab.py

```diff
@@ -241,6 +274,118 @@ def get_version_command(client, args):
+        args (Dict[str, Any]): XSOAR arguments:
+            - 'project_id' (Required): Project ID to retrieve pipeline schedules from.
+            - 'pipeline_id': ID of specific pipeline to retrieve its details.
+
+    Returns:
+        (CommandResults).
+    """
+    project_id = args.get('project_id', '')
+    pipeline_id = args.get('pipeline_id')
+    response = client.get_pipeline_request(project_id, pipeline_id)
+    response = response if isinstance(response, list) else [response]
+    outputs = [{k: v for k, v in output.items() if k in PIPELINE_FIELDS_TO_EXTRACT} for output in response]
+
+    return CommandResults(
+        outputs_prefix='GitLab.Pipeline',
+        outputs_key_field='id',
+        outputs=outputs,
+        raw_response=response,
+        readable_output=tableToMarkdown('GitLab Pipelines', outputs)
+    )
+
+
+def gitlab_jobs_list_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Returns pipeline jobs corresponding to given arguments.
+    Args:
+        client (Client): Client to perform calls to GitLab services.
+        args (Dict[str, Any]): XSOAR arguments:
+            - 'project_id' (Required): Project ID to retrieve pipeline schedules from.
+            - 'pipeline_id': ID of specific pipeline to retrieve its jobs.
+
+    Returns:
+        (CommandResults).
+    """
+    project_id = args.get('project_id', '')
+    pipeline_id = args.get('pipeline_id', '')
+    response = client.get_pipeline_job_request(project_id, pipeline_id)
+    response = response if isinstance(response, list) else [response]
+    outputs = [{k: v for k, v in output.items() if k in JOB_FIELDS_TO_EXTRACT} for output in response]
+
+    return CommandResults(
+        outputs_prefix='GitLab.Job',
+        outputs_key_field='id',
+        outputs=outputs,
+        raw_response=response,
+        readable_output=tableToMarkdown('GitLab Jobs', outputs)
+    )
+
+
+def gitlab_artifact_get_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Returns artifact corresponding to given artifact path suffix of the given job ID.
+    Args:
+        client (Client): Client to perform calls to GitLab services.
+        args (Dict[str, Any]): XSOAR arguments:
+            - 'project_id' (Required): Project ID to retrieve pipeline schedules from.
+            - 'job_id': ID of specific job to retrieve artifact from.
+            - 'artifact_path_suffix': Suffix to artifact in the artifacts directory of the job.
+
+    Returns:
+        (CommandResults).
+    """
+    project_id = args.get('project_id', '')
+    job_id = args.get('job_id', '')
+    artifact_path_suffix = args.get('artifact_path_suffix', '')
+    response = client.get_job_artifact_request(project_id, job_id, artifact_path_suffix)
+    outputs = {
+        'job_id': job_id,
+        'artifact_path_suffix': artifact_path_suffix,
+        'artifact_data': response
+    }
+
+    return CommandResults(
```

Comment:
please add a human readable

Reply:
@dantavori About that, I added it a HR at first. But the artifact data can be quite large i guess (depending on which artifact retrieved). WDYT? I will add HR that the data for artifact and given job was retrieved if length is over 100, else output data as is